### PR TITLE
Unload fixes

### PR DESCRIPTION
**Pull request by julien-thierry**
Couple of fixes for missing/failing module unloading found while running kpatch testsuite.

### CODE REVIEW

File: test/integration/centos-7/multiple.test [hunk 1]
```diff
@@ -10,9 +10,23 @@ TEST_POSTFIX="-LOADED.test"
 
 set -o errexit
 
+declare -a loaded_modules
+
+cleanup_modules() {
+	for ((idx=${#loaded_modules[@]}-1 ; idx>=0 ; idx--)); do
```
[the remark this answers is not in the document]
Comment (julien-thierry) [reply test/integration/centos-7/multiple.test]: This is to unload modules in the opposite order as the one they were loaded. I use idx so we start from the end of the list.

I agree with the idea of factorizing things for multiple.test (and potentially other tests). I'll look into that.

Thanks.

File: kpatch/kpatch [hunk 2]
```diff
@@ -415,10 +427,41 @@ case "$1" in
...
 	case "$2" in
 	"--all")
+		# Versions of linux < 5.1 livepatching require patches to be
+		# disabled in the inverse order in which they were enabled.
+		# The enabling order is not available through sysfs, and
+		# livepatchers may not have used kpatch to load modules.
+		# Just go multiple times over the list of loaded modules
+		# until were unable to disable any module remaining.
+		successful_disable="true"
+		while [ $successful_disable = "true" ]; do
```
[the remark this answers is not in the document]
Comment (julien-thierry) [reply kpatch/kpatch]: It shouldn't loop forever. As soon as there is an iteration where we were unable to disable at least one module, the loop should stop.

Unless I'm missing something.

[the remark this answers is not in the document]
Comment (sm00th) [reply kpatch/kpatch]: Ah right. The whole successful_disable thing is a bit hard to follow. Will it work if we just do `while [ 1 ]` and then change all of the `continue`s to `break`s? That will probably be easier to read.

[the remark this answers is not in the document]
Comment (joe-lawrence) [reply kpatch/kpatch]: nit: I think the commit msg captures the spirit of this comment.  My preference would be a one or two liner (if at all) and let the commit msg go into the detailed explanation.

File: kpatch/kpatch [hunk 3]
```diff
@@ -349,35 +352,44 @@ disable_patch () {
...
 			else
 				warn "retrying..."
 				sleep $RETRY_INTERVAL
 			fi
 		done
 	fi
+}
+
+disable_patch_strict () {
+	local modname=$1
```
Comment (joe-lawrence) [inline kpatch/kpatch]: nit: file convention seems to be `local modname="$1"`

File: kpatch/kpatch (hunk 2 above)
[the remark this answers is not in the document]
Comment (joe-lawrence) [reply kpatch/kpatch]: I'd be curious to see a `break` version as well.  And also, I think we've already got `while true; do` in other scripts, so to be consistent...

File: test/integration/kpatch-test [hunk 4]
```diff
@@ -117,14 +117,7 @@ log() {
...
 unload_all() {
-	for i in `/sbin/lsmod |egrep '^kpatch' |awk '{print $1}'`; do
-		if [[ $i != kpatch ]]; then
-			$KPATCH unload $i >> $LOG 2>&1 || error "\"kpatch unload $i\" failed"
-		fi
-	done
-	if /sbin/lsmod |egrep -q '^kpatch'; then
-		$RMMOD kpatch >> $LOG 2>&1 || error "\"rmmod kpatch\" failed"
-	fi
+	$KPATCH unload --all
```
Comment (joe-lawrence) [inline test/integration/kpatch-test]: The script change looks fine, but I wanted to add a nitpick on the commit message:
"The unload_all() function in kpatch-test script is out dated and does not unload patched..."  that last word should be "patches".

File: kpatch/kpatch (hunk 2 above)
[the remark this answers is not in the document]
Comment (julien-thierry) [reply kpatch/kpatch]: I've put a new version with the `while true` version. And shortened the comment.

File: test/integration/centos-7/multiple.test (hunk 1 above)
[the remark this answers is not in the document]
Comment (julien-thierry) [reply test/integration/centos-7/multiple.test]: The new version factorizes the multiple. If you can confirm whether the way is done looks ok to you or if I should do it another way.

File: kpatch/kpatch (hunk 3 above)
Comment (julien-thierry) [reply kpatch/kpatch]: Fixed in new version.

File: test/integration/kpatch-test (hunk 4 above)
Comment (julien-thierry) [reply test/integration/kpatch-test]: Fixed in new version, thanks.
